devtools/generate-appdata.py

```diff
@@ -1,23 +1,25 @@
 #!/usr/bin/env python3
 
-# This script generates a dists/scummvm.appdata.xml file with multilanguage support.
+# This script generates dists/org.scummvm.scummvm.appdata.xml file with multilanguage support.
 # The multilanguage data is extracted from po/*.po files
 
 import re
 import os
+import xml.sax.saxutils
 
-appdata_xml_template = '''<?xml version="1.0" encoding="UTF-8"?>
+APPDATA_OUTPUT_FILE = 'dists/org.scummvm.scummvm.appdata.xml'
+APPDATA_XML_TEMPLATE = '''<?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2020-2023 The ScummVM Team -->
 <component type="desktop">
-  <id>scummvm.desktop</id>
+  <id>org.scummvm.scummvm.desktop</id>
```

Comment:
```suggestion
  <id>org.scummvm.scummvm</id>
```
(Although this probably does not matter.)

Reply:
Hm, or maybe not. I am not sure whether the .desktop suffix in id is recommended or not by the latest specification. Feel free to keep it there for now.

Reply:
The spec says:
> In previous AppStream releases, the <id/> was used to associate metainfo files with their .desktop files to merge in data from .desktop files into the AppStream generator's final output. In modern metainfo files, the component-ID for desktop-application components can be an arbitrary reverse-DNS string (matching the naming rules applying to all AppStream metadata), while the [<launchable/>](https://www.freedesktop.org/software/appstream/docs/sect-Metadata-Application.html#tag-dapp-launchable) tag is used to associate .desktop files with their metainfo files.

> Note that even though the component-ID can now be any rDNS name, when updating existing applications, do not change their <id/> to drop the .desktop suffix. The rules are relaxed when picking a new component-ID for new applications, but when updating older applications they still need to keep their original <id/> (when it's otherwise compliant). The ID is used to uniquely identify applications across distributions and releases and should always remain the same for the same application. 

As we are already changing the ID to comply with rDNS only rule (which is mandatory), I removed the desktop extension.